valhalla/midgard/gridded_data.h

```diff
@@ -115,55 +115,77 @@ template <std::size_t dimensions_t> class GriddedData : public Tiles<PointLL> {
     // function stored in the array
     int case_table[3][3][3] = {
         {{0, 0, 8}, {0, 2, 5}, {7, 6, 9}},
-        {{0, 3, 4}, {1, 3, 1}, {4, 3, 0}},
+        {{0, 3, 4}, {1, 0, 1}, {4, 3, 0}},
         {{9, 6, 7}, {5, 2, 0}, {8, 0, 0}},
     };
+
+    // swap_table indicates whether the (from_pt, to_pt) segment orientation should be changed or not
+    // Lets take case_index=3: case[1][0][1] and case[1][2][1]. They correspond to the (0, -1, 0) and
+    // (0, 1, 0) sh[] values respectively. Though both cases correspond to the case_value=3 but they
+    // have different m1--m3 segment orientation.
+    /*
+     *          sh[m2]=-1        |          sh[m2]=1
+     *           /\              |              /\
+     *          /  \             |             /  \
+     *         /    \            |            /    \
+     * sh[m1]=0 ----> sh[m3]=0   |    sh[m1]=0 <---- sh[m3]=0
+     */
+    // This is needed to keep contours oriented correctly, according to the
+    // right-hand rule:
+    // "A linear ring MUST follow the right-hand rule with respect to the area it
+    // bounds, i.e., exterior rings are counterclockwise, and holes are clockwise."  (c)
+    // (c) https://tools.ietf.org/html/rfc7946#section-3.1.6
+    bool swap_table[3][3][3] = {
+        {{false, false, true}, {false, true, true}, {true, false, false}},
+        {{false, true, false}, {true, false, false}, {true, false, false}},
+        {{true, true, false}, {false, false, false}, {false, false, false}},
+    };
     std::array<std::function<void()>, 10> cases{
-        []() {},
+        [&]() { from_pt = to_pt = {}; },
```

Comment:
this is just to be more explicit i guess, since it will be skipped later?

Reply:
ah damn, I didn't notice it  :) https://github.com/valhalla/valhalla/blob/99893c90aa80857739bb812050439be1110fee22/valhalla/midgard/gridded_data.h#L270

I guess we can leave this change, anyway it doesn't change anything(as you noticed)